### PR TITLE
PWN-5418 - Implement Proto Data Store

### PR DESCRIPTION
**Pull request by Davran331122**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-5418

### CODE REVIEW

File: core/build.gradle [hunk 1]
```diff
@@ -7,7 +7,7 @@ dependencies {
     implementation 'com.github.bumptech.glide:glide:4.12.0'
 
     // https://mvnrepository.com/artifact/org.bitcoinj/bitcoinj-core
-    api 'org.bitcoinj:bitcoinj-core:0.15.10'
+    api 'org.bitcoinj:bitcoinj-core:0.16.2'
```
Comment (Davran331122) [inline core/build.gradle:10]: I Updated bitcoin library, cause it's used old Protobuf version, and we had conflicts via merge them.

File: app/src/main/java/org/p2p/wallet/infrastructure/proto/serializer/TokenKeysDataStore.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import android.content.Context
+import androidx.datastore.core.Serializer
+import androidx.datastore.dataStore
+import org.p2p.wallet.TokenKeysDataStore
+import java.io.InputStream
+import java.io.OutputStream
+
+private const val STORE_FILE_NAME = "token_keys_store.pb"
+
+class TokenKeysDataStoreSerializer(
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/infrastructure/proto/serializer/TokenKeysDataStore.kt:12]: TokenKeysDataStore

File: app/build.gradle [hunk 3]
```diff
@@ -2,6 +2,7 @@ plugins {
     id 'com.android.application'
     id 'kotlin-android'
     id 'kotlin-parcelize'
+    id "com.google.protobuf" version "0.8.17"
```
Comment (gslevinkov) [inline app/build.gradle:5]: don't forget to add
```
-keepclassmembers class * extends androidx.datastore.preferences.protobuf.GeneratedMessageLite {
    <fields>;
}
```

File: app/src/main/java/org/p2p/wallet/infrastructure/proto/serializer/TokenKeysDataStore.kt (hunk 2 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/infrastructure/proto/serializer/TokenKeysDataStore.kt:12]: and what is token keys anyway? it is public + private keys? it's Account / Wallets DataStore then

File: app/src/main/proto/token_keys_data_store.proto [hunk 4]
```diff
@@ -0,0 +1,9 @@
+syntax='proto3';
+
+option java_package = "org.p2p.wallet";
+option java_multiple_files = true;
+
+message TokenKeysDataStore {
+  string public_key = 1;
```
Comment (gslevinkov) [inline app/src/main/proto/token_keys_data_store.proto:7]: encrypted_public_key
encrypted_private_key

can we make private_key into string too?

File: app/src/main/proto/torus_data_store.proto [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+option java_package = "org.p2p.wallet";
+option java_multiple_files = true;
+
+message  TorusDataStore {
+  string ethereum_public_key = 1;
+  string mnemonic_phrase = 2;
+  string encrypted_mnemonic_phrase = 3;
+  DeviceShare device_share = 4;
+  DeviceShare custom_third_share = 5;
```
Comment (gslevinkov) [inline app/src/main/proto/torus_data_store.proto:11]: please use `optional` or `required` modifiers

**Review by gslevinkov — CHANGES_REQUESTED** [review]

**Review by kabramovp2p — APPROVED** [review]

**Pull request closed without merging**